run.go

```diff
@@ -0,0 +1,109 @@
+package testy
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gametimesf/testy/orderedmap"
+)
+
+// RunAsTest runs all registered tests under Go's testing framework. To run tests on a per-package basis, put a test
+// file in each package containing a single test that calls this function. This is recommended so accurate per-package
+// execution times are reported, as well as using the test cache. Do not import a test package into another test package
+// as that will cause the tests in the second package to get executed with the first package. If code or resources need
+// shared between test packages, put them in their own package which does not contain any test definitions.
+func RunAsTest(t *testing.T) {
+	t.Helper()
+	instance.tests.Iterate(func(pkg string, tests orderedmap.OrderedMap[string, testCase]) bool {
+		tests.Iterate(func(name string, test testCase) bool {
+			t.Run(test.Name, func(tt *testing.T) {
+				tt.Helper()
+				test.tester(tWrapper{t: tt})
+			})
+			return true
+		})
+		return true
+	})
+}
+
+// Run runs all registered tests and returns result information about them.
+//
+// TODO: ability to filter for specific packages and tests
+// TODO: channel for results to support progressive result loading?
+func Run() orderedmap.OrderedMap[string, orderedmap.OrderedMap[string, TestResult]] {
+	results := make(orderedmap.OrderedMap[string, orderedmap.OrderedMap[string, TestResult]])
+
+	instance.tests.Iterate(func(pkg string, tests orderedmap.OrderedMap[string, testCase]) bool {
+		results[pkg] = make(orderedmap.OrderedMap[string, TestResult])
+		tests.Iterate(func(name string, test testCase) bool {
+
+			res := make(chan TestResult)
+
+			wg := sync.WaitGroup{}
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				for r := range res {
+					results[pkg][r.Name] = r
+				}
+			}()
+
+			runTest(pkg, test.Name, test.tester, res)
+			close(res)
+			wg.Wait()
```

Comment:
```suggestion
			wg.Wait()
			close(res)
```

it's been a while since i've used channels but what's preventing `runTest` from writing to this closed channel? i know we want to close for the sake of the above goroutine to stop its iteration, but it appears `runTest` should add to this scope's `wg` and defer a `Done()` so we can `Wait()` on the wait group in order to protect `close(res)`. i'm sure i'll realize my misunderstanding when i look at this again.

Reply:
actually that suggestion is invalid unless `runTest` adds to this scope's wait group and the local goroutine does not.

Reply:
`runTest` runs synchronously

Reply:
specifically, that suggestion will cause the wait group to never trigger, so the code deadlocks.

Reply:
> runTest runs synchronously

🤦 duh. told you it's been a minute! 

and yes, that suggestion as it stands would cause a deadlock but i added caveats in my self-discussion.